wire-ios/Wire-iOS Tests/ConversationCreation/GroupConversationCreationCoordinatorTests.swift

```diff
@@ -0,0 +1,585 @@
+        )
+        return conversation
+    }
+}
+
+final class GroupConversationCreationCoordinatorTests: XCTestCase {
+    var sut: GroupConversationCreationCoordinator!
+    var coreDataStack: CoreDataStack!
+    var uiMOC: NSManagedObjectContext!
+    var documentsDirectory: URL?
+
+    override func setUp() {
+        super.setUp()
+        do {
+            documentsDirectory = try FileManager.default.url(for: .documentDirectory, in: .userDomainMask, appropriateFor: nil, create: true)
+        } catch {
+            XCTAssertNil(error, "Unexpected error \(error)")
+        }
+        setupCoreDataStack()
+    }
+
+    override func tearDown() {
+        super.tearDown()
+        sut?.finalize()
+        // Needs to be called before setting self.documentsDirectory to nil.
+        removeContentsOfDocumentsDirectory()
+        uiMOC = nil
+        coreDataStack = nil
+        documentsDirectory = nil
+    }
+
+    func setupCoreDataStack() {
+        let account = Account(userName: "", userIdentifier: UUID())
+        let coreDataStack = CoreDataStack(account: account,
+                                          applicationContainer: documentsDirectory!,
+                                          inMemoryStore: true)
+
+        coreDataStack.loadStores(completionHandler: { error in
+            XCTAssertNil(error)
+        })
+        self.coreDataStack = coreDataStack
+        self.uiMOC = coreDataStack.viewContext
+    }
+
+    func removeContentsOfDocumentsDirectory() {
+        do {
+            let contents = try FileManager.default.contentsOfDirectory(at: documentsDirectory!, includingPropertiesForKeys: nil, options: .skipsHiddenFiles)
+
+            for content: URL in contents {
+                do {
+                    try FileManager.default.removeItem(at: content)
+                } catch {
+                    XCTAssertNil(error, "Unexpected error \(error)")
+                }
+            }
+
+        } catch {
+            XCTAssertNil(error, "Unexpected error \(error)")
+        }
+
+    }
+
+    func testThatLoaderIsShownWhenConversationIsBeingCreated() {
+        sut = GroupConversationCreationCoordinator(creator: MockConversationCreator())
+        var showLoaderRequested = false
+        _ = sut.initialize { event in
+            switch event {
+            case .showLoader:
+                showLoaderRequested = true
+            default:
+                break
+            }
+        }
+        let userSession = MockZMUserSession()
+        let users = UserSet(arrayLiteral: MockUserType.createDefaultSelfUser(), MockUserType.createDefaultOtherUser())
+        _ = sut.createConversation(
+            withUsers: users,
+            name: "Conversation",
+            allowGuests: true,
+            allowServices: true,
+            enableReceipts: true,
+            encryptionProtocol: EncryptionProtocol.proteus,
+            userSession: userSession,
+            moc: self.uiMOC)
+        Thread.sleep(forTimeInterval: 1)
```

Comment:
What is sleep needed for?

Reply:
The insertion to the database is enqueued on `userSession` so I assume it will happen with a delay on another thread.

Reply:
Try to use `XCTAssert(waitForAllGroupsToBeEmpty(timeout: 0.5))` (you can see how it works in other tests) when you want wait for async operations.

Reply:
When I use the function (it is implemented in `ZMTBaseTest`) it fails one test (not related to the test content but to `Countly provider`

Reply:
I've been able to fix that by usage of `CoreDataFixtureTestHelper`